db/migrate/20170522104534_regroup_course_settings.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+class RegroupCourseSettings < ActiveRecord::Migration
+  def settings_key_mapping
+    @settings_key_mapping ||= {
+      announcement: :course_announcements_component,
+      forum: :course_forums_component,
+      leaderboard: :course_leaderboard_component,
+      material: :course_materials_component,
+      video: :course_videos_component,
+      virtual_classroom: :course_virtual_classrooms_component,
+      discussion_topics: :course_discussion_topics_component
+    }
+  end
+
+  def setting_keys
+    @setting_keys ||= settings_key_mapping.keys
+  end
+
+  def change
+    ActsAsTenant.without_tenant do
+      Course.all.each do |course|
+        # Retrieve all settings
+        components_settings = course.settings.components
+        sidebar_settings = course.settings.sidebar
+        course_settings = course.settings.course
+        individual_component_settings = setting_keys.map do |key|
+          [key, course.settings.public_send(key)]
+        end.to_h
+
+        # Remove unused settings, e.g. those under the key :lecture
+        course.settings = nil
```

Comment:
Would actually prefer that you remove keys to be removed (`lecture`), rename those to be renamed, and not touch the rest. This should make the migration slightly more readable.